src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/CosmosQueryNotification.cs

```diff
@@ -0,0 +1,39 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using System.Net;
+using MediatR;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Features.Storage
+{
+    public class CosmosQueryNotification : INotification
+    {
+        private Stopwatch _stopwatch;
+
+        public CosmosQueryNotification()
+        {
+            _stopwatch = Stopwatch.StartNew();
+        }
+
+        public string Operation { get; set; }
+
+        public HttpStatusCode? StatusCode { get; set; }
+
+        public double RequestCharge { get; set; }
+
+        public long LatencyMilliseconds { get; set; }
+
+        public long? CollectionSizeUsage { get; set; }
+
+        public string ResourceType { get; set; }
+
+        public void SetLatency()
```

Comment:
I would suggest decoupling the creation of this class from starting a Stopwatch. I think this should just be a payload of data.

Reply:
Moved it out to the Middleware.